src/caf/toolkit/pandas_utils/matrices.py

```diff
@@ -0,0 +1,284 @@
+    """
+
+    def __init__(
+        self,
+        matrix: pd.DataFrame,
+        *,
+        translation_factors: Optional[pd.DataFrame] = None,
+        translation_from_col: Optional[str] = None,
+        translation_to_col: Optional[str] = None,
+        translation_factors_col: Optional[str] = None,
+    ):
+
+        self._describe: pd.DataFrame = pd.DataFrame()
+
+        self.tld = None
+
+        translated_matrix: pd.DataFrame | None = None
+
+        if translation_factors is not None:
+            if (
+                (translation_factors_col is None)
+                or (translation_from_col is None)
+                or (translation_to_col is None)
+            ):
+                raise ValueError(
+                    "If translation is provided translation_from_col,"
+                    " translation_to_col and translation_factors_col "
+                    "must also be given"
+                )
+
+            self._describe["Original_Matrix"] = matrix_describe(matrix)
+
+            translated_describe_label = "Translated_Matrix"
+
+            translated_matrix = translation.pandas_matrix_zone_translation(
+                matrix,
+                translation_factors,
+                translation_from_col,
+                translation_to_col,
+                translation_factors_col,
+            )
+
+        elif (
+            (translation_factors_col is not None)
+            or (translation_from_col is not None)
+            or (translation_to_col is not None)
+        ):
+            raise ValueError(
+                "If translation_from_col,"
+                " translation_to_col or translation_factors_col are provided,"
+                " translation must also be given"
+            )
+        else:
+            translated_describe_label = "Matrix"
+        self._matrix = matrix
+        self._translated_matrix: pd.DataFrame | None = translated_matrix
+        self._describe[translated_describe_label] = matrix_describe(matrix)
+        self._distribution: cost_utils.CostDistribution | None = None
+
+    def trip_length_distribution(
+        self,
+        cost_matrix: pd.DataFrame,
+        bins: list[int],
+    ) -> None:
+        """Calculate a distribution from the matrix passed on initilisation.
+
+        Distribution is stored within the object which can be accessed using
+        the `MatrixReport.distribution` property.
+
+        Parameters
+        ----------
+        cost_matrix : pd.DataFrame
+            Cost matrix corresponding with the inputted matrix.
+        bins : list[int]
+            Bins to use for the distribution.
+        """
+        # I have ignored Mypy complaining about index/column types as this works as intended
+        try:
+            cost_matrix.index = [int(ind) for ind in cost_matrix.index]  # type: ignore[assignment]
+            cost_matrix.columns = [int(col) for col in cost_matrix.columns]  # type: ignore[assignment]
+        except ValueError:
+            pass
```

Comment:
Try using `pd.to_numeric` instead because it can apply to the whole index at once
```suggestion
        try:
            cost_matrix.index = pd.to_numeric(cost_matrix.index, downcast="integer")
            cost_matrix.columns = pd.to_numeric(cost_matrix.columns, downcast="integer")
        except ValueError:
            pass
```